src/pynxtools/nexus/nexus.py

```diff
@@ -27,16 +29,93 @@
     try_find_units,
     walk_elist,
     write_doc_string,
+    decode_or_not,
 )
 
 
+from typing import Union, Optional, Any
+import numpy as np
+
+from typing import Any, Optional
+import numpy as np
+
+from typing import Any, Optional, Union
+import numpy as np
+
+
+def decode_if_string(
+    elem: Any, encoding: str = "utf-8", decode: bool = True
+) -> Optional[Any]:
+    """
+    Decodes a numpy ndarray or list of byte objects or byte strings to strings.
+    If `decode` is False, returns the input value unchanged. Non-byte/str types
+    are returned without modification.
+
+    Args:
+        elem: A numpy ndarray, list, bytes, or any other object.
+        encoding: The encoding scheme to use. Default is "utf-8".
+        decode: A boolean flag indicating whether to perform decoding.
+
+    Returns:
+        A decoded string, a numpy array of decoded strings, a list of decoded strings,
+        or the input value if not decodable.
+        Returns None if the input is empty or invalid.
+
+    Raises:
+        ValueError: If decoding fails on bytes or numpy array elements.
+    """
+    # Early return if decoding is disabled
+    if not decode:
+        return elem
+
+    # Handle numpy arrays of bytes or strings
+    if isinstance(elem, np.ndarray):
+        if elem.size == 0:
+            return elem  # Return the empty array unchanged
+
+        if elem.dtype.kind == "S":  # Check if it's a bytes array (fixed-length strings)
+            decoded_array = np.vectorize(
+                lambda x: x.decode(encoding).rstrip("\x00")
+                if isinstance(x, bytes)
+                else x
+            )(elem)
+            return decoded_array.astype(str)  # Ensure the dtype is str
+
+        # Handle mixed-type arrays
+        if elem.dtype == object:
+            decoded_array = np.vectorize(
+                lambda x: x.decode(encoding) if isinstance(x, bytes) else x
+            )(elem)
+            return decoded_array.astype(str)  # Ensure all elements are strings
+
+    return decode_or_not(elem, encoding, decode)
+
+    # # Handle lists of bytes or strings
```

Comment:
Are arrays of strings covered? or would they be covered by the commented section?

Reply:
Since I could not just copy all the functionality to pynxtools and just call the function in the nexus_defintions, I'll ended up splitting the funcion. Bytes and list of bytes are convered by the `decode_or_not` function which comes from the dev_tools. The commented part can thus be removed.